woocommerce/woocommerce
# Add loading indicator when submitting location in Tax task

### Changes proposed in this Pull Request:

This PR is an upstream changes from https://github.com/Automattic/wc-calypso-bridge/pull/1261/files#diff-a577aa04a40191a102dfb08f7b739e30431f3b997d883de14721adcfd362c4c9R7:

1. Add a loading indicator when submitting store address in the tax task.
2. Fixes unintended reset of `isRequesting` state when updating settings
3. Add preemptive set `isRequesting` to true when optimistically updating settings to allow loading indicator

### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:


1. Go to WooCommerce > Settings
2. In General tab, remove `Address line 1`, `City`, and `Postcode / ZIP`
3. Set `Country` to `Afghanistan`
4. Click on `Save changes`
5. Go to WooCommerce > Home
1. Click `Add tax rates` task in WooCommerce Home
1. Click `WooCommerce Tax`
1. Fill in the address information but ensure country is set to `Afghanistan`
2. Click on `Continue`
1. Observe `Continue` button is changed to `busy` state as per ([screenshot](https://github.com/woocommerce/woocommerce/assets/3747241/aa75094d-fc8f-412a-b81f-e5ec63ed47d3))
3. Observe when loading finishes, it automatically moves to the second



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
